ush/forecast_postdet.sh

```diff
@@ -827,27 +469,71 @@ MOM6_out() {
   # Copy MOM_input from DATA to COM_OCEAN_INPUT after the forecast is run (and successfull)
   ${NCP} "${DATA}/INPUT/MOM_input" "${COM_CONF}/ufs.MOM_input"
 
+  # Create a list of MOM6 restart files
+  # Coarser than 1/2 degree has a single MOM restart
+  local mom6_restart_files mom6_restart_file restart_file
+  mom6_restart_files=(MOM.res.nc)
+  # 1/4 degree resolution has 4 additional restarts
+  case "${OCNRES}" in
+    "025")
+      local nn
+      for (( nn = 1; nn <= 4; nn++ )); do
+        mom6_restart_files+=("MOM.res_${nn}.nc")
+      done
+      ;;
+    *) ;;
+  esac
+
+  # Copy MOM6 restarts at the end of the forecast segment to COM for RUN=gfs|gefs
+  local restart_file
+  if [[ "${RUN}" == "gfs" || "${RUN}" == "gefs" ]]; then
+    echo "Copying MOM6 restarts for 'RUN=${RUN}' at ${forecast_end_cycle}"
+    for mom6_restart_file in "${mom6_restart_files[@]}"; do
+      restart_file="${forecast_end_cycle:0:8}.${forecast_end_cycle:8:2}0000.${mom6_restart_file}"
+      ${NCP} "${DATArestart}/MOM6_RESTART/${restart_file}" \
+             "${COM_OCEAN_RESTART}/${restart_file}"
+    done
+  fi
+
+  # Copy restarts at the beginning/middle of the next assimilation cycle to COM for RUN=gdas|enkfgdas|enkfgfs
+  if [[ "${RUN}" =~ "gdas" || "${RUN}" == "enkfgfs" ]]; then
```

Comment:
@aerorahul so this is a bit forward looking and we can take care of this seperately if neede be.  But I have an update to this section of code from the original section: 

![image](https://github.com/NOAA-EMC/global-workflow/assets/22549231/b8a42ce1-3c83-42ca-9b7b-719337e0b0e9)


Which was for the case when we start with a cold half-cycle w/out IAU and then use IAU the next full cycle.  To do this, we need to copy more restarts here.  I put the changes I made above (I haven't made a PR b/c the WCDA test has not worked in develop, but was planning to soon when I could re-test my changes).  I believe I copied what the atm model was doing in this case, but I see the atm model has changed.   So I'm wondering if this block should also change, but match closer to what the atm model is doing, which might be a few less restarts than the code updates  I have here.

Reply:
What I did in this PR is collected the `restart_date` and `restart_dir` based on `RERUN` condition, composed a list of initial conditions (and increments) based on `DOIAU`, then copy those files as part of `postdet`.

I think we can address this in a separate issue when the AOWCDA test is back up and running.  This will make it easier to insert the additional files in the right place.